ecl/hql/hqlcache.cpp

```diff
@@ -369,16 +374,49 @@ static IHqlExpression * createSimplifiedBodyDefinition(IHqlExpression * expr)
         OwnedHqlExpr newBody = createSimplifiedBodyDefinition(expr->queryChild(0)->queryBody());
         if (!newBody)
             return nullptr;
-  //MORE: check if the default arguments are simple enough (or can be simplfied)
-        return replaceChild(expr->queryBody(), 0, newBody);
+
+        IHqlExpression * params = expr->queryChild(1);
+        HqlExprArray funcArgs;
+        HqlExprArray dummyAttrs;
+        ForEachChild(i, params)
+        {
+            IHqlExpression * param = params->queryChild(i)->queryBody();
+            OwnedHqlExpr newType = createSimplifiedDefinitionType(getFullyUnqualifiedType(param->queryType()));
```

Comment:
I'm not sure why this is here.  It means that only integer parameters will be allowed.